### PR TITLE
feat(wallet-backend): Some code changes for GCP deployment 

### PR DESCRIPTION
**Pull request by cryptofan1013**
### Context
- fixes #

### Changes

### CODE REVIEW

File: docker/dev/docker-compose-gcp.yml [hunk 1]
```diff
@@ -0,0 +1,194 @@
...
+      context: ../..
+      dockerfile: ./packages/wallet-backend/Dockerfile.dev
+    depends_on:
+      - postgres
+      - rafiki-backend
+    environment:
+      NODE_ENV: development
+      PORT: 3003
+      DB_URL: postgres://wallet_backend:wallet_backend@postgres/wallet_backend
+      JWT_ACCESS_TOKEN_SECRET: u3uakiThYILFEewwhlHkepwMLM@Ikbj1LA!JQeeVyVEn$$Cj9t$$Ix0TWMYrEnpq$$X
```
Comment (dragosp1011) [inline docker/dev/docker-compose-gcp.yml:49]: I suggest to add the jwt tokens as it is done for rapyd

File: packages/wallet-frontend/package.json [hunk 2]
```diff
@@ -2,7 +2,7 @@
   "name": "wallet-frontend",
   "scripts": {
     "dev": "PORT=4003 next dev",
-    "build": "next build",
+    "build": "NEXT_PUBLIC_BACKEND_URL=http://35.196.11.156:3003 next build",
```
Comment (dragosp1011) [inline packages/wallet-frontend/package.json:5]: is this change required?

Comment (cryptofan1013) [reply packages/wallet-frontend/package.json:5]: well, the env values in docker-compose.yml wasn't affecting the build environment variables.
Which means, without this, that backend url was setting to another value.

File: docker/dev/docker-compose-gcp.yml (hunk 1 above)
Comment (beniaminmunteanu) [reply docker/dev/docker-compose-gcp.yml:49]: I don't understand :)

File: packages/wallet-frontend/package.json (hunk 2 above)
Comment (beniaminmunteanu) [reply packages/wallet-frontend/package.json:5]: I also find it a bit weird to add the env variables here in the build script

File: docker/dev/docker-compose-gcp.yml (hunk 1 above)
Comment (dragosp1011) [reply docker/dev/docker-compose-gcp.yml:49]: add them in .env file and here change it to `JWT_ACCESS_TOKEN_SECRET: ${JWT_ACCESS_TOKEN_SECRET}`

File: packages/wallet-backend/src/index.ts [hunk 3]
```diff
@@ -23,7 +23,15 @@ app.disable('X-Powered-By')
 
 app.use(
   cors({
-    origin: 'http://localhost:4003',
+    origin: [
+      'http://localhost:4003',
```
Comment (dragosp1011) [inline packages/wallet-backend/src/index.ts:27]: should these be in env file?

File: packages/wallet-frontend/package.json [hunk 4]
```diff
@@ -2,7 +2,7 @@
   "name": "wallet-frontend",
   "scripts": {
     "dev": "PORT=4003 next dev",
-    "build": "next build",
+    "build": "NEXT_PUBLIC_BACKEND_URL=https://rafiki.money/api next build",
```
Comment (dragosp1011) [inline packages/wallet-frontend/package.json:5]: the env can be set in docker on running the build script

File: docker/dev/docker-compose-gcp.yml (hunk 1 above)
Comment (cryptofan1013) [reply docker/dev/docker-compose-gcp.yml:49]: I think this can be solved after it's updated in main docker file? not docker file for gcp.

File: packages/wallet-backend/src/index.ts (hunk 3 above)
Comment (cryptofan1013) [reply packages/wallet-backend/src/index.ts]: Well, I think as the domain will be fixed, it can be here as well.

File: docker/dev/docker-compose-prod.yml [hunk 5]
```diff
@@ -0,0 +1,194 @@
...
+    container_name: wallet-frontend
+    build:
+      context: ../..
+      dockerfile: ./packages/wallet-frontend/Dockerfile-Prod.dev
+    environment:
+      NODE_ENV: development
+      PORT: 4003
+      NEXT_PUBLIC_USE_TEST_KYC_DATA: 'true'
+      NEXT_PUBLIC_BACKEND_URL: https://api.rafiki.money
+      NEXT_PUBLIC_OPEN_PAYMENTS_HOST: http://rafiki-backend
```
Comment (dragosp1011) [inline docker/dev/docker-compose-prod.yml:29]: is this value correct?

**Comment by Tymmmy** [comment]
Looks like you need to run prettier also  on auth.service file

File: docker/dev/docker-compose-prod.yml [hunk 6]
```diff
@@ -0,0 +1,194 @@
...
+    build:
+      context: ../..
+      dockerfile: ./packages/wallet-backend/Dockerfile.dev
+    depends_on:
+      - postgres
+      - rafiki-backend
+    environment:
+      NODE_ENV: development
+      PORT: 3003
+      DB_URL: postgres://wallet_backend:wallet_backend@postgres/wallet_backend
```
Comment (dragosp1011) [inline docker/dev/docker-compose-prod.yml:48]: DB urls and jwt secrets should also be in .env IMO

**Review by beniaminmunteanu — APPROVED** [review]

File: packages/wallet-backend/src/auth/auth.service.ts [hunk 7]
```diff
@@ -39,7 +39,10 @@ export const appendAccessTokenToCookie = (
   expiresIn: number
 ) => {
   res.cookie('AccessToken', token, {
-    httpOnly: true,
+    httpOnly: env.NODE_ENV === 'production',
```
Comment (dragosp1011) [inline packages/wallet-backend/src/auth/auth.service.ts:42]: extract cookies options in one common object/method

File: packages/wallet-backend/src/auth/auth.service.ts [hunk 8]
```diff
@@ -39,7 +39,10 @@ export const appendAccessTokenToCookie = (
...
+    secure: env.NODE_ENV === 'production',
+    sameSite: 'none',
+    domain: env.NODE_ENV === 'production' ? 'rafiki.money' : 'localhost:4003',
```
Comment (dragosp1011) [inline packages/wallet-backend/src/auth/auth.service.ts:45]: I think domain should be in an env variable, that will allow to setup other environments without changing the code

**Review by dragosp1011 — CHANGES_REQUESTED** [review]

File: packages/wallet-backend/src/auth/auth.service.ts (hunk 7 above)
Comment (cryptofan1013) [reply packages/wallet-backend/src/auth/auth.service.ts:42]: well, this is for wallet-backend only. In wallet-backend-update, there's only one option for cookies already.

**Review by dragosp1011 — APPROVED** [review]
